### ISSUE TITLE
Connect to SCCS using OAUTH

### ISSUE THREAD

**Issue by eschild** [SteeltoeOSS/Configuration#21]
I am able to connect to SCCS using basic auth.  However our SCCS server is being migrated to only allowing OAUTH.  Now we need to hit an endpoint with username/password to get issued a token which we can then in turn use to get the actual configs from SCCS.  Is this method supported?

**Comment by dtillman** [comment]
@eschild :  

Are you using Spring Cloud Services Config Server on Cloud Foundry (i.e. http://docs.pivotal.io/spring-cloud-services/1-5/common/) , or is this your own configuration and deployment of a Config Server?

Dave

**Comment by eschild** [comment]
It is not in Cloud Foundry.  It is our own configuration and deployment.  I did look at the code you had in Steeltoe.Extensions.Configuration.ConfigServerBase/ConfigServerConfigurationProvider.cs and modified my copy of the  GetRequestMessage function to use the overloaded HttpClientHelper.GetRequestMessage in Steeltoe.Common.Http which supports passing a function for getAccessToken so I do have a functioning system again.

**Comment by dtillman** [comment]
@eschild
We have a package Pivotal.Extensions.Configuration.ConfigServerBase that I think does what you want... It's intended to be used with Spring Cloud Services Config Server on Cloud Foundry  (i.e. http://docs.pivotal.io/spring-cloud-services/1-5/common/) which does the same thing... protects the config server endpoint by requiring an Access token.
I think you could just use that if you want.

Also, if you don't want to use it .. have a look in https://github.com/SteeltoeOSS/Common/blob/master/src/Steeltoe.Common.Http/HttpClientHelper.cs, as there are some helper methods  that are used by Pivotal.Extensions.Configuration.ConfigServerBase to get the access token, etc.  ... they may be of help also.

Dave

**Comment by eschild** [comment]
@dtillman 
Thanks for pointing me at the Pivotal.Extensions.Configuration.ConfigServerBase package.  I was not aware of it.  In the end I decided not to use it but I did model from it to build a provider that implements STCS.ConfigServerConfigurationProvider and then just overrides GetRequestMessage.  Thank you for your  quick responses and support with this issue. Your assistance is greatly appreciated.
